README.md

```diff
@@ -1,137 +1,326 @@
 
+``` cmake
+generate_parameter_library(
+minimal_node_parameters # cmake target name for the parameter library
+src/minimal_node_parameters.yaml # path to input yaml file
+include/minimal_node/validators.hpp # Optional path to header containing custom validators
+)
+```
+The generated library then needs to be linked to the executable.
+```cmake
+target_link_libraries(minimal_node PRIVATE
+  rclcpp::rclcpp
+  rclcpp_lifecycle::rclcpp_lifecycle
+  minimal_node_parameters
+)
+```
+The above cmake is substituted in place of `# additional libraries ...`. Notably, the third argument of `generate_parameter_library` specifies a user provided header to include in the generated code. All validation functions defined in this header can be referred to in the YAML validation dictionary.
 
-you will get the error
+## Integrate generated struct into project source code
+Only a few source code changes are needed to integrate the generated parameter library code. First, the library must be included via `#include "minimal_node_parameters.hpp"`. Note that the include name is exactly the same as the generated library name specified for in cmake.
+```c++
+#include "minimal_node_parameters.hpp"
 ```
-terminate called after throwing an instance of 'rclcpp::exceptions::ParameterUninitializedException'
-  what():  parameter 'command_interfaces' is not initialized
-[ros2run]: Aborted
+In the initialization code, an instance of `ParamListener` and `Params` should be created. At this time, parameters will be declared and read. An exception will be thrown if any validation fails or any required parameters were not set.
+```c++
+param_listener = std::make_shared<minimal_node::ParamListener>(node);
+params_ = param_listener->get_params();
 ```
+Finally, the instance of `Params` can be updated dynamically during the main loop as follows.
+```c++
+if (param_listener->is_old(params_)) {
+  params_ = param_listener->get_params();
+}
+```
+Substituting in all changes, the new source is shown below.
+```c++
+#include "rclcpp/rclcpp.hpp"
+#include "minimal_node_parameters.hpp"
+
+class MinimalInterface {
+public:
+    MinimalInterface(rclcpp::Node::SharedPtr node) {
+        param_listener = std::make_shared<minimal_node::ParamListener>(node);
+        params_ = param_listener->get_params();
+    }
+    void update(const std_msgs::msg::String::SharedPtr msg) {
+        if (param_listener->is_old(params_)) {
+          params_ = param_listener->get_params();
+        }
+      // ...
+    }
+
+    std::share_ptr<minimal_node::ParamListener> param_listener;
+    minimal_node::Params params_;
+
+};
+
+int main(int argc, char * argv[])
+{
+  rclcpp::init(argc, argv);
+  node = std::make_shared<rclcpp::Node>("minimal_node");
+  auto minimal_interface = MinimalInterface(node);
+
+  while(rclcpp::ok()){
+    minimal_interface.update();
+    // ...
+  }
+
+  return 0;
+}
+
+```
```

Comment:
Because of the example project, I think this minimal example code in the README is too much and should be removed.